src/Test/TestCountTracker.java

```diff
@@ -0,0 +1,72 @@
+package Test;
+
+import Engine.UnitMods.CountTracker;
+
+public class TestCountTracker extends TestCase
+{
+  @Override
+  public boolean runTest()
+  {
+    boolean testPassed = true;
+    testPassed &= validate(testNull(), " CountTracker null test failed");
+    testPassed &= validate(testStress(), " CountTracker stress test failed");
+    return testPassed;
+  }
+
+  private boolean testNull()
+  {
+    boolean testPassed = true;
+
+    CountTracker<Integer, Double> ct = new CountTracker<>();
+
+    try
+    {
+      ct.resetCountFor(null);
+      // Run these operations a few times to check for statefulness
+      for( int i = 0; i < 3; ++i )
+      {
+        ct.hasCountFor(null);
+        ct.getCountFor(null);
+        ct.getCountFor(null, null);
+        ct.getCountFor(0, null);
+        ct.getCountFor(null, 0.0);
+        ct.incrementCount(null, null);
+        ct.incrementCount(0, null);
+        ct.incrementCount(null, 0.0);
```

Comment:
What is this testing? Just that you can call each of these functions without it crashing? No verifying that return values are correct, or that the 0/null counts actually increment?

Reply:
It's testing that `CountTracker` is null-safe.
As for correctness, I don't know and don't really care what's correct.
That may change in the future, but for now there's no defined "correct" behavior for these cases - is `null` a valid key? I don't know.